src/generate.ts

```diff
@@ -0,0 +1,65 @@
+import { generateClientTypes } from "./generateClientTypes";
+import { generatePreviewTypes } from "./generatePreviewTypes";
+import { generateVisibilityMapType } from "./generateVisibilityTypes";
+import { extractProperties } from "./helpers";
+import { WidgetXml } from "./WidgetXml";
+
+const mxExports = [
+    "ActionValue",
+    "DynamicValue",
+    "EditableValue",
+    "FileValue",
+    "ListValue",
+    "NativeIcon",
+    "NativeImage",
+    "ObjectItem",
+    "WebIcon",
+    "WebImage",
+];
+
+export function generateForWidget(widgetXml: WidgetXml, widgetName: string) {
+    if (!widgetXml?.widget?.properties) {
+        throw new Error("[XML] XML doesn't contains <properties> element");
+    }
+    if (widgetXml.widget.$.pluginWidget !== "true") {
+        throw new Error("[XML] Attribute pluginWidget=true not found. Please review your XML");
+    }
+
+    const isNative = widgetXml.widget.$.supportedPlatform === "Native";
+
+    const properties = (widgetXml.widget.properties.length > 0
+        ? extractProperties(widgetXml.widget.properties[0])
+        : []
+    ).filter((prop) => prop && prop.$ && prop.$.key);
+
+    const clientTypes = generateClientTypes(widgetName, properties, isNative);
+    const modelerTypes = generatePreviewTypes(widgetName, properties);
+    const visibiltyMap = generateVisibilityMapType(properties);
```

Comment:
Isn't this replaced with conditional properties?

Reply:
aand its gone